ui/components/src/ContentWithLegend/model/content-with-legend-model.ts

```diff
@@ -113,15 +128,30 @@ export function getContentWithLegendLayout({
   }
 
   const { position } = legendOptions;
-
-  const legendWidth = position === 'Right' ? 200 : width;
-
-  // TODO: account for number of legend items returned when adjusting legend spacing
-  let legendHeight = LEGEND_HEIGHT_SM;
-  if (position === 'Right') {
-    legendHeight = height;
-  } else if (height >= PANEL_HEIGHT_LG_BREAKPOINT) {
-    legendHeight = LEGEND_HEIGHT_LG;
+  const mode = getLegendMode(legendOptions.mode);
+
+  let legendWidth;
+  let legendHeight;
+
+  if (mode === 'List') {
+    // TODO: normalize list to share similar height options as the table
+    // when we add more size options.
+    legendWidth = position === 'Right' ? 200 : width;
+
+    // TODO: account for number of legend items returned when adjusting legend spacing
```

Comment:
Logic for the "list" mode is largely unchanged besides the vars being declared above.